### ISSUE TITLE
Let's encrypt missing info on error

### ISSUE THREAD

**Issue by deimosfr** [containous/traefik#2396]
### Do you want to request a *feature* or report a *bug*?

feature

### What did you do?

I configured Traefik on Kubernetes with let's encrypt and Cloudflare API. I've 3 FQDN for the moment to declare to Treafik. For all of them it's ok, but for one specific FQDN, it tooks several hours (~ 1 day) to be active.

I tried to find the issue but did not get enough details (even in debug mode) in the logs to figure out what's wrong. Here is an example of what I've got:
```
Error getting ACME certificates [my.failed.fqdn] : Cannot obtain certificates map[my.failed.fqdn:Error presenting token: Cloudflare API Error 
legolog: 2017/11/08 09:18:06 [INFO][my.failed.fqdn] AuthURL: https://acme-v01.api.letsencrypt.org/acme/authz/BRmkwz5gNUUSvHfAkTjX5qxt_wd5kBiua6hhkmxFtUw
legolog: 2017/11/08 09:18:06 [INFO][my.failed.fqdn] acme: Trying to solve DNS-01
[INFO][my.failed.fqdn] acme: Obtaining bundled SAN certificate
```

### What did you expect to see?

I'm expecting to see more information on the Cloudflare API error. Impossible here to understand what's wrong.

### What did you see instead?

The certificate couldn't be generated. I had to wait several hours but it doesn't looks like to be an API limitation. I just don't have enough information at that point.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      v1.4.2
Codename:     roquefort
Go version:   go1.9.2
Built:        2017-11-02_01:47:46PM
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
    checkNewVersion = false
    IdleTimeout = "180s"
    MaxIdleConnsPerHost = 500
    logLevel = "INFO"
    defaultEntryPoints = ["http", "https"]

    [retry]
    attempts = 3

    [web]
    address = ":8081"

    [kubernetes]
    endpoint = "http://localhost:8080"

    [consul]
    endpoint = "consul:8500"
    watch = true
    prefix = "traefik"

    [acme]
    email = "username@myfqdn"
    storage = "traefik/acme/account"
    entryPoint = "https"
    OnHostRule = true
    onDemand = true
    acmeLogging = true
    dnsProvider = "cloudflare"
    delaydontcheckdns = 20
    #caServer = "https://acme-staging.api.letsencrypt.org/directory"

    [[acme.domains]]
      main = "first.domain.com"
    [[acme.domains]]
      main = "second.domain.com"
    [[acme.domains]]
      main = "third.domain.com"

    [entryPoints]
      [entryPoints.http]
      address = ":80"
      compress = true
        [entryPoints.http.redirect]
        entryPoint = "https"
      [entryPoints.https]
      address = ":443"
        [entryPoints.https.tls]
```

**Comment by Thom-x** [comment]
I have exaclty the same error.

**Comment by nmengin** [comment]
Hello @deimosfr .

Please, can you check if the mandatories environment variables are correctly set : https://docs.traefik.io/configuration/acme/#dnsprovider ?

Thanks.

**Comment by deimosfr** [comment]
Hi @nmengin,

I confirm that CLOUDFLARE_EMAIL and CLOUDFLARE_API_KEY are set and used.
It wouldn't work for 2 others FQDN if it wasn't the case. In addition when it was working for 2 of 3, I tested with other FQDN and it was working.

So ONLY for a specific FQDN, during ~1 day, it wasn't working without any comprehensive error messages and worked without doing something specific right after.

Thanks

**Comment by nmengin** [comment]
@deimosfr 

I'm going to investigate the interactions between Træfik and Lego in the way to check if we don't hide error messages from LE or if the problem is in Lego.

I'll give a feedback about  this point when I'll have more information.

**Comment by deimosfr** [comment]
Thanks !

**Comment by nmengin** [comment]
Hello @deimosfr,

I analyzed deeper the problem and apparently it comes from the the Json decoding of the body contained in the reponse when the challenge is presented to Let's Encrypt.
As you can see [here](https://github.com/xenolf/lego/blob/master/providers/dns/cloudflare/cloudflare.go#L209), it's difficult to get more information on the cause of the error but, I opened a [PR](https://github.com/xenolf/lego/pull/452) on LEGO repo in the way to add the body response into the error message.

I hope it will be accepted and it can be helpful.

**Comment by deimosfr** [comment]
Thanks a lot @nmengin, looks got. Hope it will be soon merged

**Comment by ldez** [comment]
Closed by #3895
